File: api/src/main/java/net/blueshell/api/service/CommitteeService.java
package net.blueshell.api.service;

import net.blueshell.api.base.BaseModelService;
import net.blueshell.api.model.Committee;
import net.blueshell.api.repository.CommitteeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CommitteeService extends BaseModelService<Committee, Long, CommitteeRepository> {

    @Autowired
    public CommitteeService(CommitteeRepository repository, ApplicationEventPublisher events) {
        super(repository, events);
    }

    public List<Committee> findALlByUserId(Long id) {
        return repository.findALlByMembersUserIdEquals(id);
    }
}

File: api/src/main/kotlin/net/blueshell/api/service/CommitteeService.kt
package net.blueshell.api.service

import net.blueshell.api.model.committee.Committee
import net.blueshell.api.repository.committee.CommitteeRepository
import net.blueshell.api.service.base.BaseModelService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class CommitteeService @Autowired constructor(
    repository: CommitteeRepository
) : BaseModelService<Committee, Long, CommitteeRepository>(repository) {
    fun findAllByUserId(id: Long): MutableList<Committee> {
        return repository.findAllByUserId(id) as MutableList<Committee>
    }
}
